src/Parallel/CharmRegistration.hpp

```diff
@@ -152,13 +152,13 @@ struct RegisterParallelComponent : RegistrationHelper {
   ~RegisterParallelComponent() override = default;
 
   void register_with_charm() const noexcept override {
-    static bool done_registration{false};
-    if (done_registration) {
+    static std::string parallel_component_name{};
+    if (not parallel_component_name.empty()) {
```

Comment:
do you really need to have this guard here? I thought `register_parallel_components` already guards against multiple registration. If that guard isn't working I think we need to double check that there aren't threading issues here

Reply:
I don't understand the question. The old code had a guard using a `bool`, the new code needs a `string` so it just uses `string.empty()` to replace the bool in the existing guard. Are you questioning whether the old coded needed a guard at all? I can change the to preserve the `bool` guard if that's helpful in understanding things.

Reply:
ah sorry, I missed that. I think leaving as-is should be fine. I don't think the check needs to be done at all because the `register_parallel_components` should only be run once. I'm happy to defer testing this to an issue :)

Reply:
Got it.. I'll revert this back to the bool guard so if we make future changes then all the different registrations will be identical and easy to handle. I created issue #3218